packages/web/app/src/components/organization/settings/oidc-integration-section.tsx

```diff
@@ -1042,15 +1203,30 @@ function UpdateOIDCIntegrationForm(props: {
                     }}
                     tabIndex={0}
                   >
-                    Close
+                    Cancel
                   </Button>
                   <Button type="submit" disabled={oidcUpdateMutation.fetching}>
-                    Save
+                    Save Properties
                   </Button>
                 </div>
               </div>
             </form>
           </div>
+          <div className="space-y-1 text-sm font-medium leading-none">
+            <p>Default Resource Assignments</p>
+            <p className="text-muted-foreground text-xs font-normal leading-snug">
+              This permitted resources for new members who sign in via OIDC.{' '}
+              <span className="font-medium">Only members with the Admin role can modify it.</span>
+            </p>
+          </div>
+          <div className="pb-4">
+            <OIDCDefaultResourceSelector
+              oidcIntegrationId={props.oidcIntegration.id}
+              disabled={!props.isAdmin}
+              organization={props.organization}
+              resourceAssignment={props.oidcIntegration.defaultResourceAssignment ?? {}}
+            />
+          </div>
```

Comment:
`Organization.projects` is filtered based on what the viewer can see. In order to work around this we could do the following:

Add `Organization.projectsForResourceSelector` -> Returns the Projects in a minimal form based on what is needed for the resource selector to work.

```graphql
type ProjectForResourceSelector {
  id: ID!
  slug: String!
  targets: [TargetForResourceSelector!]!
}

type TargetForResourceSelector {
  id: ID!
  slug: String!
  services: [String!]!
  appDeployments: [String!]!
}
```

Reply:
If we don't do this, we might have a weird scenario where assigned resources "vanish", after someone without all project access edits this list.

Reply:
It could then simply check if the viewer has the permissions for the contexts where it is used e.g.`oidc:modify` and `member:xxxx`